src/inet/mqtt_client.c

```diff
@@ -604,12 +610,40 @@ static int handle_publish(struct mqtt_client_t *self_p,
     }
 
     topic[topic_size] = '\0';
+    qos = ((flags >> 1) & 0x3);
+
+
+    log_object_print(self_p->log_object_p,
+                     LOG_DEBUG,
+                     OSTR("QoS: %d, Flags: %x.\r\n"),
+                     qos,
+                     flags);
+
+    if (qos == 0) {
+        payload_size = (size - topic_size - 2);
+    } else {
+        if (chan_read(self_p->transport.in_p, buf, 2) != 2) { /* read the packet identifier */
```

Comment:
Move the comment to the line before the read. Also write comments as normal sentences, starting with a captital letter and ending with a period.